src/file/validator.ts

```diff
@@ -20,3 +21,46 @@ export const schema = Joi.object<ConfigurationFile>().keys({
     .optional()
     .allow(null),
 });
+
+/** Evaluate if the regex expression inside a configuration are valid.
+ * @returns a tuple of type [boolean, string]. If the boolean is false, the string will contain an error message
+ * @example
+ * const [result, error] = validateRegularExpressions(myConfig);
+ * if (!result) {
+ *   throw new Error(error);
+ * } else {
+ *   runExpression(myConfig);
+ * }
+ */
+export const validateRegularExpressions = (
+  config: ConfigurationFile,
+  logger: ActionLogger,
+): [true] | [false, string] => {
+  /** Regex evaluator */
+  const isRegexValid = (regex: string): boolean => {
+    try {
+      new RegExp(regex);
+      return true;
+    } catch (e) {
+      logger.error(e as Error);
+      return false;
+    }
+  };
+
+  for (const rule of config.rules) {
+    for (const condition of rule.condition.include) {
+      if (!isRegexValid(condition)) {
+        return [false, `Include condition '${condition}' is not a valid regex`];
+      }
+    }
+    if (rule.condition.exclude) {
+      for (const condition of rule.condition.exclude) {
+        if (!isRegexValid(condition)) {
+          return [false, `Exclude condition '${condition}' is not a valid regex`];
+        }
+      }
+    }
+  }
+
+  return [true];
+};
```

Comment:
I could maybe change this to be the following:
```suggestion
export const validateRegularExpressions = (
  config: ConfigurationFile,
  logger: ActionLogger,
): boolean => {
  /** Regex evaluator */
  const isRegexValid = (regex: string): boolean => {
    try {
      new RegExp(regex);
      return true;
    } catch (e) {
      logger.error(e as Error);
      return false;
    }
  };

  for (const rule of config.rules) {
    for (const condition of rule.condition.include) {
      if (!isRegexValid(condition)) {
        logger.error(`Include condition '${condition}' is not a valid regex`);
        return false;
      }
    }
    if (rule.condition.exclude) {
      for (const condition of rule.condition.exclude) {
        if (!isRegexValid(condition)) {
          logger.error(`Exclude condition '${condition}' is not a valid regex`);
          return false;
        }
      }
    }
  }
  
  return true;
};
```

what do you think?